addon/record-data.js

```diff
@@ -581,6 +581,10 @@ export default class M3RecordData {
 
   removeFromRecordArrays() {
     if (CUSTOM_MODEL_CLASS) {
+      // No need to remove us from record arrays if the store is beeing destroyed
+      if (this.globalM3CacheRD._storeIsDestroying) {
+        return;
+      }
```

Comment:
This seems like an odd place to stash a store destroying flag. Does the store wrapper not know whether the store is being destroyed?

Reply:
No, there is no public API for it, so I made one we create. Any ideas for a better place to put it?

Reply:
putting the flag on the record data as a workaround seems fine to me as long as we agree that this is a deficiency in the public API that should be sorted out

Reply:
Moved to using schema's `isDestroying` flag